lib/geo.js

```diff
@@ -10,23 +10,31 @@ var assert = require('assert');
  */
 
 exports.nearFilter = function nearFilter(where) {
+  var keyPath = [];
   var result = false;
 
-  if (where && typeof where === 'object') {
-    Object.keys(where).forEach(function(key) {
-      var ex = where[key];
-
-      if (ex && ex.near) {
-        result = {
-          near: ex.near,
-          maxDistance: ex.maxDistance,
-          unit: ex.unit,
-          key: key,
-        };
-      }
-    });
+  searchForNear(where);
+
+  function searchForNear(node) {
+    if (node && typeof node === 'object') {
+      Object.keys(node).forEach(function(key) {
+        var ex = node[key];
```

Comment:
dunno - I was going to change the name to something more meaningful but I didn't want to trigger a possible reason to get the PR rejected... I'm starting to exceed the amount of company time I can spend on this PR since we have a major client project that's behind schedule :/ - let me know if you have a better name, and I'll update the PR.